src/components/icon/icon.registry.ts

```diff
@@ -28,44 +43,52 @@ export class IconsRegistry {
     this.listeners.delete(callback);
   }
 
-  private parseSVG(svgString: string): ParsedIcon {
-    const parsed = this._parser.parseFromString(svgString, 'image/svg+xml');
-    const svg = parsed.querySelector('svg');
-
-    if (parsed.querySelector('parsererror') || !svg) {
-      throw new Error('SVG element not found or malformed SVG string.');
+  public setRefsByTheme(theme: Theme) {
+    if (this.theme !== theme) {
+      this.theme = theme;
+
+      for (const { alias, target } of iconReferences) {
+        const external = this.references
+          .get(alias.collection)
+          ?.get(alias.name)?.external;
+
+        this.setIconRef({
+          alias,
+          target: target.get(this.theme) ?? target.get('default')!,
+          overwrite: !external,
+        });
```

Comment:
Doubt it'll be a significant difference atm (there are pretty much 17 default icons in iconReferences), but this could use some optimization to check if the target actually changes to avoid running `notifyAll` for all entries - either here or in general built into `setIconRef`